test/fake-node.js

```diff
@@ -91,9 +94,16 @@ FakeNode.prototype._registerEndpoints = function _registerEndpoints() {
         var path = self.endpoints[endpointType].path;
 
         self.channel.register(path, function handleRequest(req, res, arg2, arg3) {
+            self.endpoints[endpointType].handler(req, res, arg2, arg3);
+
+            // After we have handled the request and sent the response, we emit
+            // the request so that the validate function can store and digested
+            // it. We want to emit this after, and not before the response has
```

Comment:
Why do we want to handle the request first, and then call the test handler? Because that's how we write the tests: we poke SUT, SUT reacts, we assert. If you look at the history of this code, you'll see the opposite: we poke SUT, and this happen in parallel:

1. We react (call the test handler).
2. SUT reacts by, e.g., sending messages from the first reaction.